src/main/kotlin/io/ionic/libs/ionfilesystemlib/helper/IONFILEContentHelper.kt

```diff
@@ -0,0 +1,227 @@
+package io.ionic.libs.ionfilesystemlib.helper
+
+import android.content.ContentResolver
+import android.database.Cursor
+import android.net.Uri
+import android.provider.DocumentsContract
+import android.provider.MediaStore
+import android.provider.OpenableColumns
+import io.ionic.libs.ionfilesystemlib.helper.common.FILE_MIME_TYPE_FALLBACK
+import io.ionic.libs.ionfilesystemlib.helper.common.readByChunks
+import io.ionic.libs.ionfilesystemlib.helper.common.readFull
+import io.ionic.libs.ionfilesystemlib.model.IONFILEExceptions
+import io.ionic.libs.ionfilesystemlib.model.IONFILEFileType
+import io.ionic.libs.ionfilesystemlib.model.IONFILEMetadataResult
+import io.ionic.libs.ionfilesystemlib.model.IONFILEReadInChunksOptions
+import io.ionic.libs.ionfilesystemlib.model.IONFILEReadOptions
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.catch
+import kotlinx.coroutines.flow.flow
+import kotlinx.coroutines.flow.flowOn
+import kotlinx.coroutines.withContext
+import java.io.File
+import java.io.FileNotFoundException
+import java.io.FileOutputStream
+
+internal class IONFILEContentHelper(private val contentResolver: ContentResolver) {
+
+    /**
+     * Reads contents of a file using content resolver
+     *
+     * @param uri the content uri to read the file from
+     * @param options options for reading the file
+     * @return success with file contents string if it was read successfully, error otherwise
+     */
+    suspend fun readFile(
+        uri: Uri,
+        options: IONFILEReadOptions
+    ): Result<String> = withContext(Dispatchers.IO) {
+        runCatching {
+            val fileContents: String = contentResolver.openInputStream(uri)?.use { inputStream ->
+                inputStream.readFull(options)
+            } ?: throw IONFILEExceptions.UnknownError()
+            return@runCatching fileContents
+        }.mapError()
+    }
+
+    /**
+     * Reads the contents of a file in chunks
+     *
+     * Useful when the file does not fit in entirely memory.
+     *
+     * @param uri the content uri to read the file from
+     * @param options options for reading the file in chunks; refer to [IONFILEReadInChunksOptions]
+     * @return a (cold) flow in which the chunks are emitted;
+     * the flow completes after all chunks are emitted (unless an error occurs somewhere in-between)
+     */
+    fun readFileInChunks(
+        uri: Uri,
+        options: IONFILEReadInChunksOptions
+    ): Flow<String> = flow {
+        contentResolver.openInputStream(uri)?.use { inputStream ->
+            inputStream.readByChunks(
+                options,
+                DEFAULT_BUFFER_SIZE,
+                onChunkRead = { chunk -> emit(chunk) }
+            )
+        } ?: throw IONFILEExceptions.UnknownError()
+    }.catch { throw it.mapError() }
```

Comment:
Do we need to run .catch { throw it.mapError() } inside of the IO thread? Probably not right? So we could do:

```
flow {
    // read file in chunks
}.flowOn(Dispatchers.IO)
.catch { throw it.mapError() }
```